### DISCUSSION TITLE
PHP Error on Object-Oriented File Upload Class

### DISCUSSION THREAD

**Discussion by ZeppiPurple** [community/community#142877] [Programming Help]
### Body

This is my code

<?php
include './global-management.php';

list('user_id' => $user_id, 'username' => $username, 'conn' => $conn) = setUpDatabaseAndValidateSession();

header('Content-Type: application/json');

function getSystemStats() {
    $cpuUsage = sys_getloadavg()[0];
    $ramTotalGB = round(shell_exec('free | grep Mem | awk \'{print $2}\'') / 1024 / 1024, 2);
    $ramUsedGB = round(shell_exec('free | grep Mem | awk \'{print $3}\'') / 1024 / 1024, 2);
    $ramFreeGB = round(shell_exec('free | grep Mem | awk \'{print $4}\'') / 1024 / 1024, 2);

    return [
        'cpu' => round($cpuUsage * 100),
        'ram' => [
            'total' => $ramTotalGB,
            'used' => $ramUsedGB,
            'free' => $ramFreeGB
        ]
    ];
}

echo json_encode(getSystemStats());
createLogbookEntry($user_id, "Retrieved system statistics", $conn); // Logbook entry
?>

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by EliasDeHondt** [comment]
@ZeppiPurple we meet again haha

The first line of your code appears to have a syntax error due to improper assignment. It seems you're trying to destructure an array returned from `setUpDatabaseAndValidateSession()`, but it’s not correctly formatted. You should assign it to an array first or use a different method for handling the return values.
```php
list($user_id, $username, $conn) = setUpDatabaseAndValidateSession();
```

> Please consider what you’re doing in this code. It seems a bit unusual to collect statistics from a Linux machine in this manner. Just something to keep in mind.

**Reply by ZeppiPurple** [reply]
It works now, thank you for helping me this time too! And I'll try to find another way to collect statistics. Thanks again @EliasDeHondt

**Comment by ebndev** [comment]
Please see [this comment](https://github.com/orgs/community/discussions/142857#discussioncomment-11078706) for reference to why this discussion was closed and locked. Thanks!
